linkerd/app/src/config.rs

```diff
@@ -0,0 +1,1058 @@
+const DEFAULT_INBOUND_CONNECT_TIMEOUT: Duration = Duration::from_millis(100);
+const DEFAULT_INBOUND_CONNECT_BACKOFF: ExponentialBackoff = ExponentialBackoff {
+    min: Duration::from_millis(100),
+    max: Duration::from_millis(500),
+    jitter: 0.1,
+};
+const DEFAULT_OUTBOUND_DISPATCH_TIMEOUT: Duration = Duration::from_secs(3);
+const DEFAULT_OUTBOUND_CONNECT_TIMEOUT: Duration = Duration::from_secs(1);
+const DEFAULT_OUTBOUND_CONNECT_BACKOFF: ExponentialBackoff = ExponentialBackoff {
+    min: Duration::from_millis(100),
+    max: Duration::from_millis(500),
+    jitter: 0.1,
+};
+const DEFAULT_DNS_CANONICALIZE_TIMEOUT: Duration = Duration::from_millis(100);
+const DEFAULT_RESOLV_CONF: &str = "/etc/resolv.conf";
+
+/// It's assumed that a typical proxy can serve inbound traffic for up to 100 pod-local
+/// HTTP services and may communicate with up to 10K external HTTP domains.
+const DEFAULT_INBOUND_ROUTER_CAPACITY: usize = 100;
+const DEFAULT_OUTBOUND_ROUTER_CAPACITY: usize = 10_000;
+
+const DEFAULT_INBOUND_ROUTER_MAX_IDLE_AGE: Duration = Duration::from_secs(60);
+const DEFAULT_OUTBOUND_ROUTER_MAX_IDLE_AGE: Duration = Duration::from_secs(60);
+
+// 10_000 is arbitrarily chosen for now...
+const DEFAULT_INBOUND_MAX_IN_FLIGHT: usize = 10_000;
+const DEFAULT_OUTBOUND_MAX_IN_FLIGHT: usize = 10_000;
+
+const DEFAULT_DESTINATION_GET_SUFFIXES: &str = "svc.cluster.local.";
+const DEFAULT_DESTINATION_PROFILE_SUFFIXES: &str = "svc.cluster.local.";
+
+const DEFAULT_IDENTITY_MIN_REFRESH: Duration = Duration::from_secs(10);
+const DEFAULT_IDENTITY_MAX_REFRESH: Duration = Duration::from_secs(60 * 60 * 24);
+
+// By default, we keep a list of known assigned ports of server-first protocols.
+//
+// https://www.iana.org/assignments/service-names-port-numbers/service-names-port-numbers.txt
+const DEFAULT_PORTS_DISABLE_PROTOCOL_DETECTION: &[u16] = &[
+    25,   // SMTP
+    587,  // SMTP
+    3306, // MySQL
+];
+
+const INBOUND_CONNECT_BASE: &str = "INBOUND_CONNECT";
+const OUTBOUND_CONNECT_BASE: &str = "OUTBOUND_CONNECT";
+
+// === impl Config ====
+
+impl<O: OrigDstAddr> Config<O> {
+    pub fn with_orig_dst_addrs_from<P: OrigDstAddr>(self, orig_dst_addrs: P) -> Config<P> {
+        Config {
+            outbound: self
+                .outbound
+                .with_orig_dst_addrs_from(orig_dst_addrs.clone()),
+            inbound: self.inbound.with_orig_dst_addrs_from(orig_dst_addrs),
+            dns: self.dns,
+            identity: self.identity,
+            destination: self.destination,
+            admin: self.admin,
+            tap: self.tap,
+            trace_collector: self.trace_collector,
+        }
+    }
+}
+
+impl Config {
+    pub fn try_from_env() -> Result<Self, EnvError> {
+        Self::parse(&Env)
+    }
+
+    /// Load a `Config` by reading ENV variables.
+    pub fn parse<S: Strings>(strings: &S) -> Result<Self, EnvError> {
+        // Parse all the environment variables. `parse` will log any errors so
```

Comment:
Not a blocker, really, but it seems to me like this massive function out to be split into smaller `parse(strings: &S)` methods on the various `FooConfig` structs?

Reply:
that's a yak for another day